yang/vyatta-protocols-frr-isis-v1.yang

```diff
@@ -0,0 +1,822 @@
+      description "Define the style of TLVs metric supported.";
+      configd:help "Define the style of TLVs metric supported.";
+    }
+
+    leaf lsp-mtu {
+      type uint16 {
+        range "128..4352";
+      }
+      configd:help "MTU of an LSP.";
+      description "MTU of an LSP.";
+    }
+  }
+
+	grouping redistribute-protocol {
+		leaf kernel {
+      type level;
+      must ". != \"isis\"";
+      configd:help "Kernel routes (not installed via the zebra RIB).";
+      description "Kernel routes (not installed via the zebra RIB).";
+    }
+    leaf connected {
+      type level;
+      must ". != \"isis\"";
+      configd:help "Connected routes (directly attached subnet or host).";
+      description "Connected routes (directly attached subnet or host).";
+    }
+    leaf static {
+      type level;
+      must ". != \"isis\"";
+      configd:help "Statically configured routes.";
+      description "Statically configured routes.";
+    }
+		leaf ospf {
+      type level;
+      must ". != \"isis\"";
+      configd:help "Open Shortest Path First (OSPFv2).";
+      description "Open Shortest Path First (OSPFv2).";
+    }
+		leaf bgp {
+      type level;
+      must ". != \"isis\"";
+      configd:help "Redistribute bgp routes";
+      description "Redistribute bgp routes";
+    }
+		leaf rip {
+      type level;
+      must ". != \"isis\"";
+      configd:help "Redistribute rip routes";
+      description "Redistribute rip routes";
+    }
+  }
+
+	grouping isis-redistribute {
+    container redistribute {
+      configd:help "Redistributes routes learned from other routing protocols.";
+      description "Redistributes routes learned from other routing protocols.";
+      container ipv4 {
+        configd:help "IPv4 route redistribution.";
+        description "IPv4 route redistribution.";
+        uses redistribute-protocol;
+      }
+      container ipv6 {
+        configd:help "IPv6 route redistribution.";
+        description "IPv6 route redistribution.";
+        uses redistribute-protocol;
+      }
+    }
+  }
+
+  grouping isis-parameters-all {
+    uses isis-redistribute;
+    uses isis-net;
+    uses isis-area-domain-password;
+  }
+
+  augment /protocols:protocols {
+    list isis {
+      configd:help "Intermediate System to Intermediate System (ISIS) parameters";
+      description "Intermediate System to Intermediate System (ISIS) parameters";
+      must "(as-num != 23456)" {
+        error-message "Invalid AS Number, 23456 (Reserved)";
+      }
+      must "count(../isis/as-num) <= 1" {
+        error-message "Another ISIS instance is already running" ;
+      }
+      key "as-num";
+      leaf as-num {
+        description "ISIS AS Number";
+          type uint32 {
+            range 1..max;
+          }
+          configd:help "ISIS parameters";
+      }				
+      uses isis-redistribute;
+      uses isis-net;
+      uses isis-area-domain-password;
+
+    }
+  }
+
+  grouping itf-ip-router-isis {
+    container router {
```

Comment:
`router` is quite a generic container name. If we define this in the IS-IS module then another feature wanting to augment its own nodes into the `router` container would have to depend on IS-IS.

For that reason I would suggest something like:
```
container isis {
    ...
    leaf instance {
        ...
    }
}
```
Does that sound ok?

Reply:
Yes, you are right. 
This is Resolved.